src/Components/WebAssembly/WebAssembly.Authentication/src/IAccessTokenProviderAccessor.cs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Authentication.Internal
+{
+    /// <summary>
+    /// This is an internal API that supports the Microsoft.AspNetCore.Components.WebAssembly.Authentication
+    /// infrastructure and not subject to the same compatibility standards as public APIs.
+    /// It may be changed or removed without notice in any release.
+    /// </summary>
+    public interface IAccessTokenProviderAccessor
+    {
+        /// <summary>
+        /// This is an internal API that supports the Microsoft.AspNetCore.Components.WebAssembly.Authentication
+        /// infrastructure and not subject to the same compatibility standards as public APIs.
+        /// It may be changed or removed without notice in any release.
+        /// </summary>
+        IAccessTokenProvider TokenProvider { get; }
```

Comment:
I'm not really certain what this is used for. I can only spot where it's used in tests.

What am I missing? Or is this really only for unit testing?

Reply:
This is to solve a cyclical dependency in the design. The AuthenticationService is the one handling all the communication with JavaScript to retrieve tokens from the underlying JS layer as well as handling the auth flows and updating the user. 

I wanted to separate the piece where the user is mapped from the results of the interop call to the ClaimsPrincipal (which is the user factory). As a result, the AuthenticationService has a dependency on the UserFactory, which gets injected.

That in turn, prevents users from injecting an `IAccessTokenProvider` in their implementation since that creates a cyclical dependency. After thinking about it, I decided that the best way to support this scenario (which I think it is important) and to avoid doing something more complicated or not as intuitive was to solve the problem in the base class.

The idea is that UserFactory injects the IAccessTokenProviderAccessor and that it exposes the ITokenProvider as a property that gets computed lazily. That has two main effects:
* Users creating a custom factory don't have to figure out the details as they are forced to provide a suitable constructor when extending the class.
* User's don't run into a weird DI error when they try to inject an IAccessTokenProvider instance themselves, since they don't have the need because the base class already offers it for convenience.

The accessor breaks the cyclical dependency and remains a small implementation detail.

Reply:
Would it be reasonable then to make `IAccessTokenProvider` a regular `public` and supported API? It's a really simple interface. Is there any harm in supporting it?

I know we prefer not to make things public if unnecessary, but it seems like this does potentially show up in user code  ("they are forced to provide a suitable constructor when extending the class") so it's only fair to make it public.

Reply:
I'm going to tackle this as a potential change in the post API review cleanups